nibabel/cifti2/cifti2.py

```diff
@@ -0,0 +1,848 @@
+                         'CIFTI_STRUCTURE_CORTEX_LEFT',
+                         'CIFTI_STRUCTURE_CORTEX_RIGHT',
+                         'CIFTI_STRUCTURE_DIENCEPHALON_VENTRAL_LEFT',
+                         'CIFTI_STRUCTURE_DIENCEPHALON_VENTRAL_RIGHT',
+                         'CIFTI_STRUCTURE_HIPPOCAMPUS_LEFT',
+                         'CIFTI_STRUCTURE_HIPPOCAMPUS_RIGHT',
+                         'CIFTI_STRUCTURE_OTHER',
+                         'CIFTI_STRUCTURE_OTHER_GREY_MATTER',
+                         'CIFTI_STRUCTURE_OTHER_WHITE_MATTER',
+                         'CIFTI_STRUCTURE_PALLIDUM_LEFT',
+                         'CIFTI_STRUCTURE_PALLIDUM_RIGHT',
+                         'CIFTI_STRUCTURE_PUTAMEN_LEFT',
+                         'CIFTI_STRUCTURE_PUTAMEN_RIGHT',
+                         'CIFTI_STRUCTURE_THALAMUS_LEFT',
+                         'CIFTI_STRUCTURE_THALAMUS_RIGHT')
+
+
+def _value_if_klass(val, klass, none_ok=True):
+    if none_ok and val is None:
+        return val
+    elif isinstance(val, klass):
+        return val
+    else:
+        raise ValueError('Not a valid %s instance.' % klass.__name__)
+
+
+def _value_or_make_klass(val, klass):
+    if val is None:
+        return klass()
+    else:
+        return _value_if_klass(val, klass)
+
+
+def _underscore(string):
+    """ Convert a string from camelcase to underscored """
+    string = re.sub(r'([A-Z]+)([A-Z][a-z])', r'\1_\2', string)
+    return re.sub(r'([a-z0-9])([A-Z])', r'\1_\2', string).lower()
+
+
+class Cifti2MetaData(xml.XmlSerializable):
+    """ A list of key-value pairs stored in the list self.data """
+
+    def __init__(self, nvpair=None):
+        self.data = []
+        self.add_metadata(nvpair)
+
+    def _add_remove_metadata(self, metadata, func):
+        pairs = []
+        if isinstance(metadata, dict):
+            pairs = metadata.items()
+        elif isinstance(metadata, (list, tuple)):
+            if not isinstance(metadata[0], string_types):
+                for item in metadata:
+                    self._add_remove_metadata(item, func)
+                return
+            elif len(metadata) == 2:
+                pairs = [tuple((metadata[0], metadata[1]))]
+            else:
+                raise ValueError('nvpair must be a 2-list or 2-tuple')
+        else:
+            raise ValueError('nvpair input must be a list, tuple or dict')
+
+        for pair in pairs:
+            if func == 'add':
+                if pair not in self.data:
+                    self.data.append(pair)
+            elif func == 'remove':
+                self.data.remove(pair)
+            else:
+                raise ValueError('Unknown func %s' % func)
+
+    def add_metadata(self, metadata):
+        """Add metadata key-value pairs
+
+        This allows storing multiple keys with the same name but different
+        values.
+
+
+        Parameters
+        ----------
+        metadata : 2-List, 2-Tuple, Dictionary, List[2-List or 2-Tuple]
```

Comment:
Parameter description should be one self-contained line.   Maybe `metadata : sequence length 2 or mapping or sequence of sequences`